src/libcore/sync/atomic.rs

```diff
@@ -923,13 +923,14 @@ impl<T> AtomicPtr<T> {
         unsafe { atomic_swap(self.p.get() as *mut usize, ptr as usize, order) as *mut T }
     }
 
-    /// Stores a value into the pointer if the current value is the same as the `current` value.
+    /// Stores a value into the pointer if the current pointer *address* value is the same
+    /// as the `current` pointer *address* value.
     ///
-    /// The return value is always the previous value. If it is equal to `current`, then the value
-    /// was updated.
+    /// The return value is always the previous value contained in the pointer. If it is equal
+    /// to the value pointed by `current`, then the value was updated.
```

Comment:
I'd still like the wording to be refined here per my previous comments. Might also be good to have some more eyes on this, since it's hard to phrase well with the distinctions between different pointers and values being made. cc @rust-lang/libs